internal/utils/cleanup/cleanup.go

```diff
@@ -0,0 +1,20 @@
+package cleanup
+
+// Cleanup provides an easy way to clean up resources after an operation fails.
+type Cleanup struct {
+	fns []func()
+}
+
+func (c *Cleanup) Add(fn func()) {
+	c.fns = append(c.fns, fn)
```

Comment:
what are we cleaning here?

Reply:
It's just a nice utility thing when you're doing something that might need to be cleaned up if the process fails.

So for example, when creating a stack branch, we do a sequence of operations that should be cleaned up if any of the subsequent operations fail. Right now I think it's mostly just reverting the checkout. You could also write it using just defer but it's a little jankier.

```go
// using this Cleanup utility
var cu cleanup.Cleanup
... = repo.Checkout(...)
cu.Add(func() { repo.Checkout(originalBranch) })

// do more stuff

// on success, we cancel the cleanup so they don't run
cu.Cancel()
return ...
```
versus
```go
// using this Cleanup utility
didSucceed := false
... = repo.Checkout(...)
defer func() {
    if didSucceed { return }
    repo.Checkout(originalBranch)
}()

// do more stuff

// on success, we cancel the cleanup so they don't run
didSucceed = true
return ...
```